File: pinyougou-web/pinyougou-user-web/src/main/webapp/js/controller/userController.js
/** 定义控制器层 */
app.controller('userController', function($scope, $timeout, baseService){

    // 定义json对象
    $scope.user = {}

    // 用户注册
    $scope.save = function () {

        // 判断密码是否一致
        if ($scope.okPassword && $scope.user.password == $scope.okPassword) {
            // 发送异步请求
            baseService.sendPost("/user/save?code=" + $scope.code,
                $scope.user).then(function (response) {
                // 获取响应数据
                if (response.data) {
                    // 跳转到登录页面
                    // 清空表单数据
                    $scope.user = {};
                    $scope.okPassword = "";
                    $scope.code = "";
                } else {
                    alert("注册失败！");
                }
            });
        } else {
            alert("两次密码不一致！");

        }

    };

    // 判断验证码和短信验证码
    $scope.test = function () {

        // 判断密码是否一致
        if ($scope.okPassword && $scope.user.password == $scope.okPassword) {
            // 发送异步请求
            baseService.sendPost("/user/save?code=" + $scope.code,
                $scope.user).then(function (response) {
                // 获取响应数据
                if (response.data) {
                    // 跳转到登录页面
                    // 清空表单数据
                    $scope.user = {};
                    $scope.okPassword = "";
                    $scope.code = "";
                } else {
                    alert("注册失败！");
                }
            });
        } else {
            alert("两次密码不一致！");

        }

    };







            // 获取登录用户名
            $scope.showName = function () {
                baseService.sendGet("/user/showName").then(function (response) {
                    // 获取响应数据
                    $scope.loginName = response.data.loginName;
                });
            };


            //显示手机号码
            $scope.showPhone = function () {
                baseService.sendPost("/user/findUser").then(function (response) {
                    $scope.phone = response.data.phone;
                })

            };

            // 定义显示文本
            $scope.tipMsg = "获取短信验证码";
            $scope.flag = false;

            // 根据后端手机号码查询,发送短信验证码
            $scope.sendSmsCode1 = function () {

                baseService.sendPost("/user/findUser").then(function (response) {
                    $scope.phone = response.data.phone;
                })

                // 判断手机号码的有效性
                if ($scope.phone && /^1[3|4|5|7|8|9]\d{9}$/.test($scope.phone)) {
                    // 发送异步请求
                    baseService.sendGet("/user/sendSmsCode?phone="
                        + $scope.phone
                    ).then(function (response) {
                        // 获取响应数据
                        if (response.data) {
                            // 倒计时 (扩展)
                            $scope.flag = true;
                            // 调用倒计时方法
                            $scope.downcount(90);
                        } else {
                            alert("获取短信验证码失败！");
                        }
                    });
                } else {
                    alert("手机号码不正确！");
                }
            };


            //密码设置(修改)
            $scope.update = function () {
                // 判断密码是否一致
                if ($scope.confirm_password && $scope.user.password == $scope.confirm_password) {
                    // 发送异步请求
                    baseService.sendPost("/user/update",
                        $scope.user).then(function (response) {
                        // 获取响应数据
                        if (response.data) {
                            // 跳转到首页
                            // 清空表单数据
                            $scope.user = {};
                            $scope.user.username = "";
                            $scope.confirm_password = "";
                            alert("设置成功!")
                            //$scope.code = "";
                        } else {
                            alert("设置失败！");
                        }
                    });
                } else {
                    alert("两次密码不一致！");
                }
            };


            // 定义显示文本
            $scope.tipMsg = "获取短信验证码";
            $scope.flag = false;

            // 根据前端手机号码查询,发送短信验证码
            $scope.sendSmsCode = function () {

                // 判断手机号码的有效性
                if ($scope.user.phone && /^1[3|4|5|7|8|9]\d{9}$/.test($scope.user.phone)) {
                    // 发送异步请求
                    baseService.sendGet("/user/sendSmsCode?phone="
                        + $scope.user.phone).then(function (response) {
                        // 获取响应数据
                        if (response.data) {
                            // 倒计时 (扩展)
                            $scope.flag = true;
                            // 调用倒计时方法
                            $scope.downcount(90);
                        } else {
                            alert("获取短信验证码失败！");
                        }
                    });
                } else {
                    alert("手机号码不正确！");
                }
            };


            // 倒计时方法
            $scope.downcount = function (seconds) {
                if (seconds > 0) {
                    seconds--;
                    $scope.tipMsg = seconds + "秒，后重新获取";

                    // 开启定时器
                    $timeout(function () {
                        $scope.downcount(seconds);
                    }, 1000);
                } else {
                    $scope.tipMsg = "获取短信验证码";
                    $scope.flag = false;
                }
            };


});

File: pinyougou-web/pinyougou-user-web/src/main/java/com/pinyougou/user/controller/UserController.java
package com.pinyougou.user.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.User;
import com.pinyougou.service.UserService;
import org.springframework.security.core.context.SecurityContext;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.*;

import javax.servlet.http.HttpServletRequest;

/**
 * 用户控制器
 *
 * @author lee.siu.wah
 * @version 1.0
 * <p>File Created at 2019-03-16<p>
 */
@RestController
@RequestMapping("/user")
public class UserController {

    @Reference(timeout = 10000)
    private UserService userService;


    /** 用户注册 */
    @PostMapping("/save")
    public boolean save(@RequestBody User user, String code){
        try{

            // 检验短信验证码
            boolean flag = userService.checkSmsCode(user.getPhone(), code);
            if (flag) {
                userService.save(user);
            }
            return flag;
        }catch (Exception ex){
            ex.printStackTrace();
        }
        return false;
    }

    /** 第一步短信验证码校验 */
    @PostMapping("/testCode")
    public boolean testCode( String phone, String code){
        try{

            // 检验短信验证码
            boolean flag = userService.testCode(phone, code);
            if (flag) {
                return flag;
            }

        }catch (Exception ex){
            ex.printStackTrace();
        }
        return false;
    }




    /** 第二步短信验证码校验,更改号码 */
    @PostMapping("/updatePhone")
    public boolean updatePhone(@RequestBody User user, String code, HttpServletRequest request){
        try{
            String username = request.getRemoteUser();
            // 检验短信验证码
            boolean flag = userService.checkSmsCode(user.getPhone(), code);
            if (flag) {
                userService.updatePhone(username,user.getPhone());
            }
            return flag;
        }catch (Exception ex){
            ex.printStackTrace();
        }
        return false;
    }








    /** 密码设置 */
    @PostMapping("/update")
    public boolean update(@RequestBody User user){
        try{
            userService.update(user);

            return true;
        }catch (Exception ex){
            ex.printStackTrace();
        }
        return false;
    }


    /** 发送短信验证码 */
    @GetMapping("/sendSmsCode")
    public boolean sendSmsCode(String phone){
        try{
            return userService.sendSmsCode(phone);
        }catch (Exception ex){
            ex.printStackTrace();
        }
        return false;
    }

    /** 获取登录用户名 */
    @PostMapping("/findUser")
    public User findUser(){
       try {
           SecurityContext context = SecurityContextHolder.getContext();
           String loginName = context.getAuthentication().getName();
           User user = new User();
           user.setUsername(loginName);
           User user1 = userService.findUser(user).get(0);
           System.out.println(user1.getPhone());
           return user1;

       }catch (Exception e){
           throw new RuntimeException(e);
       }

    }



}
